android/build.gradle

```diff
@@ -1,30 +1,77 @@
 apply plugin: 'com.android.library'
 
 def safeExtGet(prop, fallback) {
-    rootProject.ext.has(prop) ? rootProject.ext.get(prop) : fallback
+  rootProject.ext.has(prop) ? rootProject.ext.get(prop) : fallback
+}
+
+buildscript {
+    // The Android Gradle plugin is only required when opening the android folder stand-alone.
+    // This avoids unnecessary downloads and potential conflicts when the library is included as a
+    // module dependency in an application project.
+    if (project == rootProject) {
+        repositories {
+            mavenCentral()
+            google()
+        }
+        dependencies {
+            classpath 'com.android.tools.build:gradle:3.2.1'
+        }
+    }
+
 }
 
 android {
-    compileSdkVersion safeExtGet('compileSdkVersion', 25)
+    compileSdkVersion safeExtGet('compileSdkVersion', 28)
+    buildToolsVersion safeExtGet('buildToolsVersion', "28.0.3")
+    lintOptions {
+       abortOnError false
+    }
+    // Used to override the NDK path/version on internal CI or by allowing
+    // users to customize the NDK path/version from their root project (e.g. for M1 support)
+    if (rootProject.hasProperty("ndkPath")) {
+        ndkPath rootProject.ext.ndkPath
+    }
+    if (rootProject.hasProperty("ndkVersion")) {
+        ndkVersion rootProject.ext.ndkVersion
+    }
 
     defaultConfig {
-        minSdkVersion 16
-        targetSdkVersion 22
-        versionCode 2
-        versionName "1.1"
-        ndk {
-            abiFilters "armeabi-v7a", "x86"
-        }
+      minSdkVersion safeExtGet('minSdkVersion', 16)
+      targetSdkVersion safeExtGet('targetSdkVersion', 28)
+      versionCode 1
+      versionName "1.0"
     }
 }
 
 repositories {
-    maven { url 'https://zendesk.jfrog.io/zendesk/repo' }
+    mavenLocal()
+    exclusiveContent {
```

Comment:
only way I could guarantee it didn't try to fetch react-native from maven

Reply:
Not sure if related but this is how we do it in the WorkerApp per comment from githu

Reply:
Take a look at the linked github issue from react native

Reply:
yeah but since this is a linked package we don't want to read from the package.json for the version, since this is a module it should use the local version, this is how it's set in `react-native-svg` as well as `react-native-async-storage`